lib/rights_api/query.rb

```diff
@@ -1,30 +1,34 @@
 # frozen_string_literal: true
 
+require "cgi"
+
+require_relative "error"
+require_relative "query_parser"
 require_relative "result"
 require_relative "services"
 
 module RightsAPI
   class Query
-    DEFAULT_LIMIT = 1000
-    attr_reader :table_name
+    attr_reader :model, :params
 
-    # @param name [String, Symbol] The name of the table to be queried.
-    def initialize(table_name:)
-      @table_name = table_name
+    # @param model [String, Symbol] The name of the table to be queried.
+    def initialize(model:, params: {})
+      @params = params.is_a?(String) ? CGI.parse(params) : params
```

Comment:
We're already doing `CGI.parse(params)` here https://github.com/hathitrust/rights_api/pull/8/files#diff-453fd6c2f36cfaa9cbd1ab6e7ca014eba2ff36da0489ac3cf83e2381a1bdb0cfR30 so this doesn't seem necessary (and there's no reason this class should care that it could be running under a web server..?)

Reply:
That was something I had in an experimental version since before the unit/integration split, somehow I envisioned testing this with a long list of URLs or API paths. It is indeed not needed here. Removed in new commit with a minor fix for the docstring which is out of date.